### PR TITLE
Add SourceLocator for Sequence

### PR DESCRIPTION
**Pull request by unlsycn**
### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- Bugfix

#### Desired Merge Strategy

- Rebase: You will rebase the PR onto master and it will be merged with a merge commit.

#### Release Notes

Add source locator for Sequence

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.6.x`, `5.x`, or `6.x` depending on impact, API modification or big change: `7.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

**Comment by sequencer** [comment]
This is not a fix bug a feature, please amend your commit.

File: src/main/scala/chisel3/ltl/LTL.scala [hunk 1]
```diff
@@ -257,7 +257,7 @@ object Sequence {
     *
     * See `SequenceAtom` for more details.
     */
-  def apply(atoms: SequenceAtom*): Sequence = {
+  def apply(atoms: SequenceAtom*)(implicit sourceInfo: SourceInfo): Sequence = {
```
Comment (sequencer) [inline src/main/scala/chisel3/ltl/LTL.scala:260]: Should we use macro for this? wdyt @jackkoenig

**Comment by unlsycn** [comment]
> This is not a fix bug a feature, please amend your commit.

Done.

File: src/main/scala/chisel3/ltl/LTL.scala (hunk 1 above)
Comment (jackkoenig) [reply src/main/scala/chisel3/ltl/LTL.scala:260]: Class `Sequence` does not define an `apply` method, so fortunately there is no need!

**Review by jackkoenig — APPROVED** [review]

**Pull request merged**